translation-service-eTranslation/src/main/java/eu/europeana/api/translation/service/eTranslation/ETranslationTranslationService.java

```
@@ -0,0 +1,273 @@
+    this.credentialUsername=username;
+    this.credentialPwd=password;
+  }
+	
+  @Override
+  public void translate(List<TranslationObj> translationObjs) throws TranslationException {    
+    List<String> externalRefsForCallbacks = new ArrayList<>();//this variable below captures only the external references of successfully created requests
+    List<String> externalRefsAll = new ArrayList<>();
+    List<Integer> indexesTranslated = new ArrayList<>();
+    //create eTransaltion requests and send them
+    try {
+      for(int i=0;i<translationObjs.size();i++) {
+        String sourceLang = translationObjs.get(i).getSourceLang();
+        String targetLang = translationObjs.get(i).getTargetLang();
+        String text = translationObjs.get(i).getText();
+        if(sourceLang!=null && targetLang!=null && !StringUtils.isBlank(text)) {
+          String extRef = generateExternalReferenceAndSaveToMap();
+          externalRefsAll.add(extRef);
+          String reponseCode=null;
+          if(!StringUtils.isBlank(baseUrl)) {
+            String requstBody = createTranslationBody(text,sourceLang,targetLang,extRef);
+            /*
+             * For simulating the eTranslation callbacks for the local testing, please use the method 
+             * createHttpRequestSimulator(extRef), instead of the createHttpRequest(requestBody), like this:
+             * String reponseCode = createHttpRequestSimulator(extRef, text);
+             * and comment out the last catch block (catch (IOException e)). The simulator method executes 
+             * in a new thread and calls back the same method as the eTranslation callback.
+             */
+            reponseCode = createHttpRequest(requstBody);
+          }
+          else {
+            reponseCode = createHttpRequestSimulator(extRef, text);
+          }
+           
+          //in case of the successfull request, the return code number is positive, otherwise negative  
+          if(Integer.valueOf(reponseCode) >= 0) {
```

in the case that the request was not successfully registered, we need to throw an exception